os_migrate/plugins/module_utils/resource.py

```diff
@@ -228,6 +233,18 @@ def _needs_update(self, target):
         """
         return self._data_without_info() != target._data_without_info()
 
+    # Not meant to be overriden in majority of subclasses.
+    def _remove_readonly_params(self, sdk_params):
+        """Remove readonly parameters from the `sdk_params` collection that
+        cannot be used for update.
+
+        Returns: filtered dict of parameters
+        """
+        for name in self.readonly_sdk_params:
+            if name in sdk_params:
+                sdk_params.pop(name)
+        return sdk_params
```

Comment:
Since i know you're a Rust fan -- if we were writing this in Rust, it would outright force us to do (1) or even stricter variant of (2) with full deep copy and not reusing any internal data. What you've implemented here, or the "weak" copy variant i suggested as (2), would be refused by the compiler.

That makes me second guess my preference for (2) :smile: but i'm still fine with either (1) or (2).

Reply:
Or full deep copy is fine too of course.

Reply:
I think option 1 is more simple, easier to read.  not sure if option two is worth the effort at this time.